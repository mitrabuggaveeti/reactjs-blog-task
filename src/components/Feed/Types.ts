export interface InewState {
    post : {
        data : {
            userId: number
            id: number
            title: string
            body: string
        }[],
        error : string
    }
}