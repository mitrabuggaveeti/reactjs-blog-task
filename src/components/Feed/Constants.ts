export const WAIT_RESPONSE = 'Loading';
export const NO_DATA = 'No Posts available'