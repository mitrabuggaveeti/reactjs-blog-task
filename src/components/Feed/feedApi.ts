export const getAllPosts = async () =>{

    try {
        const response = await fetch(`https://jsonplaceholder.typicode.com/posts`);
        const data = await response.json();

        if(response.ok){
            return {
                data,
                error : ''
            }
        }
        else{
            return {
                data : [],
                error : 'Failed in fetching posts'
            };
        }
        
    } catch (error) {
        return {
            data : [],
            error 
        };
    }
}

