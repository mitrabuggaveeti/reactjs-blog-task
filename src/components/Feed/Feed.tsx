import React,{useEffect,useState} from 'react'
import Post from '../Post/Post'
import {getAllPosts} from './feedApi';
import {InewState} from './Types';

const Feed : React.FC = () => {

    const [posts, setPosts] = useState<InewState['post']>({data : [], error : 'Loading'});

    useEffect(() => {
        const getData = async() =>{
            const data = await getAllPosts();
            setPosts(data)
        }

        getData();
    }, [])

    const allPosts = posts.data.slice(0,20).map((post) => <Post post={post} key={post.id} posts={posts} setPosts={setPosts}/>);

    return (
        <div data-testid="feed">
            {!posts.error ? allPosts : <h1>{posts.error}......</h1>}
        </div>
    )
}

export default Feed
