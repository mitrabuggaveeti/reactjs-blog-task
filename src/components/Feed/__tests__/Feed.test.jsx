import { cleanup, render,waitFor} from "@testing-library/react";
import Feed from "../Feed";


jest.mock('../feedApi');

describe('Feed component should', () => {
    afterEach(cleanup);

    it('Should have two blogs rendered',async () => {
                
        const { getByTestId, asFragment } = render(<Feed />);

        const listNode = await waitFor(() => getByTestId('feed'));
        expect(listNode.children).toHaveLength(2);
        expect(asFragment()).toMatchSnapshot();
    });

})
