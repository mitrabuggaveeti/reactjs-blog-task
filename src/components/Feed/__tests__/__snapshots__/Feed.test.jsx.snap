// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed component should Should have two blogs rendered 1`] = `
<DocumentFragment>
  <div
    data-testid="feed"
  >
    <div
      class="card"
    >
      <div
        class="container"
      >
        <div
          class="title"
        >
          <strong>
             test 
          </strong>
        </div>
        <div
          class="description"
        >
          test
        </div>
        <div
          class="btns"
        >
          <button
            class="editBtn"
          >
            Edit
          </button>
          <button
            class="deleteBtn"
            data-testid="deleteComments"
          >
            Delete
          </button>
          <button
            class="commentsBtn"
            data-testid="showComments"
          >
            Show Comments
          </button>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="container"
      >
        <div
          class="title"
        >
          <strong>
             test 
          </strong>
        </div>
        <div
          class="description"
        >
          test
        </div>
        <div
          class="btns"
        >
          <button
            class="editBtn"
          >
            Edit
          </button>
          <button
            class="deleteBtn"
            data-testid="deleteComments"
          >
            Delete
          </button>
          <button
            class="commentsBtn"
            data-testid="showComments"
          >
            Show Comments
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
